resources/manager/factory_aws_test.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/utils/testhelper"
+	agentconfig "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type awsTestFetcherConfig struct {
+	fetching.AwsBaseFetcherConfig `config:",inline"`
+}
+
+type awsTestFetcher struct {
+	resourceCh chan fetching.ResourceInfo
+	cfg        awsTestFetcherConfig
+}
+
+func (f *awsTestFetcher) Fetch(ctx context.Context, cMetadata fetching.CycleMetadata) error {
+	f.resourceCh <- fetching.ResourceInfo{
+		Resource:      awsTestResource{AwsConfig: f.cfg.AwsConfig},
+		CycleMetadata: cMetadata,
+	}
+
+	return nil
+}
+
+func (f awsTestFetcher) Stop() {
+}
+
+type awsTestResource struct {
+	AwsConfig aws.ConfigAWS
+}
+
+func (a awsTestResource) GetMetadata() fetching.ResourceMetadata {
+	return fetching.ResourceMetadata{}
+}
+
+func (a awsTestResource) GetData() any {
+	return a.AwsConfig
+}
+
+func (a awsTestResource) GetElasticCommonData() any {
+	return nil
+}
+
+type awsTestFactory struct{}
+
+func (n *awsTestFactory) Create(log *logp.Logger, c *agentconfig.C, ch chan fetching.ResourceInfo) (fetching.Fetcher, error) {
+	cfg := awsTestFetcherConfig{}
+	err := c.Unpack(&cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &awsTestFetcher{ch, cfg}, nil
+}
+
+func awsMockedFetcherConfig(s *FactoriesTestSuite, awsConfig aws.ConfigAWS) *agentconfig.C {
+	c := agentconfig.NewConfig()
+	conf := config.DefaultConfig
+	conf.Type = config.InputTypeEKS
+	err := c.Merge(awsConfig)
+	s.Nil(err)
+
+	return c
+}
+
+func (s *FactoriesTestSuite) TestCreateFetcherWithAwsCredentials() {
+	var tests = []struct {
+		key       string
+		awsConfig aws.ConfigAWS
+	}{
+		{"some_fetcher",
+			aws.ConfigAWS{
+				AccessKeyID:     "key",
+				SecretAccessKey: "secret",
+				SessionToken:    "session",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		s.F.ListFetcherFactory(test.key, &awsTestFactory{})
+		c := awsMockedFetcherConfig(s, test.awsConfig)
+
+		f, err := s.F.CreateFetcher(s.log, test.key, c, s.resourceCh)
+		s.NoError(err)
+		err = f.Fetch(context.TODO(), fetching.CycleMetadata{})
+		results := testhelper.CollectResources(s.resourceCh)
+
+		s.Equal(1, len(results))
+		s.Nil(err)
+
+		result := results[0].GetData().(aws.ConfigAWS)
+		s.Equal(test.awsConfig.AccessKeyID, result.AccessKeyID)
+		s.Equal(test.awsConfig.SecretAccessKey, result.SecretAccessKey)
+		s.Equal(test.awsConfig.SessionToken, result.SessionToken)
+	}
+}
+
+func (s *FactoriesTestSuite) TestRegisterFetchersWithAwsCredentials() {
+	var tests = []struct {
+		key       string
+		awsConfig aws.ConfigAWS
+	}{
+		{"some_fetcher",
+			aws.ConfigAWS{
+				AccessKeyID:     "key",
+				SecretAccessKey: "secret",
+				SessionToken:    "session",
+			},
+		},
+		//{"another_fetcher",
+		//	aws.ConfigAWS{
+		//		AccessKeyID:     "new_key",
+		//		SecretAccessKey: "new_secret",
+		//		SessionToken:    "new_session",
+		//	},
+		//},
```

Comment:
remove

Reply:
I kept the content - just removed the `//` :)